src/hashset.c

```diff
@@ -1317,24 +1319,29 @@ size_t hashsetScan(hashset *t, size_t cursor, hashsetScanFunction fn, void *priv
                 in_probe_sequence |= b->everfull;
             }
 
-            /* Iterate over indices in larger table that are the expansion of
-             * the index pointed to by the cursor in the smaller table. */
-            do {
-                /* Emit elements in table 1. */
-                bucket *b = &t->tables[table1][cursor & mask1];
-                for (int pos = 0; pos < ELEMENTS_PER_BUCKET; pos++) {
-                    if (b->presence & (1 << pos)) {
-                        void *emit = emit_ref ? &b->elements[pos] : b->elements[pos];
-                        fn(privdata, emit);
+            if ((growing && !ahead_rehash) || (!growing && !behind_rehash)) {
```

Comment:
A similar comment please, to explain: already rehashed source or not yet rehashed destination.

Reply:
I determined that this shortcut is invalid - see my other long comment with the ascii pictures 😅